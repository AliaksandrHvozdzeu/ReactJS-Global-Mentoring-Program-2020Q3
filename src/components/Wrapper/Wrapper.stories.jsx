import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import Wrapper from './js/Wrapper';
import './css/Wrapper.css';
import '../../App.css';
import MovieAddForm from "../MovieAdd/js/MovieAddForm";

export default {
  title: 'Wrapper',
  decorators: [withKnobs],
};

const children = (
  <MovieAddForm onSubmit={() => {alert("closeAction")}}
                modalTitle={"modalTitle"}
                closeAction={() => {alert("closeAction")}}
                initialState={[]}
                availableGenres={['All', 'Documentary', 'Comedy', 'Horror', 'Crime']}/>
);

export const Default = () => (
  <Wrapper children={children} />
);
