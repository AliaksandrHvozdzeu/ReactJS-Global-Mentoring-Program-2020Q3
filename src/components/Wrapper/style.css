.wrapper {
    margin: 0 6%;
    position: relative;
    height: 100%;
    width: 88%;
    overflow: hidden;
}

.error {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--white);
    font-size: 3rem;
}


.unselectable {
    -webkit-touch-callout: none; /* iOS Safari */
    -webkit-user-select: none; /* Chrome/Safari/Opera */
    -khtml-user-select: none; /* Konqueror */
    -moz-user-select: none; /* Firefox */
    -ms-user-select: none; /* Internet Explorer/Edge */
    user-select: none;
    /* Non-prefixed version, currently
                           not supported by any browser */
}
