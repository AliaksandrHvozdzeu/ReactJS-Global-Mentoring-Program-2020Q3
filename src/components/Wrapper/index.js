import Wrapper from './js/Wrapper';

export default Wrapper;
