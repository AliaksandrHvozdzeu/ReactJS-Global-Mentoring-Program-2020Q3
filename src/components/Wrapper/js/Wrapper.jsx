import React from 'react';
import PropTypes from 'prop-types';
import '../css/Wrapper.css';

const Wrapper = ({ children }) => {
  return <div className="wrapper unselectable">{children}</div>;
};

Wrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Wrapper;
