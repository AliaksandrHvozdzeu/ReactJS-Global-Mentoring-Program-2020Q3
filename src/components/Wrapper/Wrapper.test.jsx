import React from 'react';
import rerender from 'react-test-renderer';
import { render } from 'enzyme';
import Wrapper from './js/Wrapper';

describe('when Wrapper rendered', () => {
  test('then child component should be rendered', () => {
    const TestChildComponent = <div className="test">Test child element</div>;
    const component = rerender.create(<Wrapper>{TestChildComponent}</Wrapper>);
    const tree = component.toJSON();
    const wrapperComponent = render(<Wrapper>{TestChildComponent}</Wrapper>);
    expect(wrapperComponent.find('.wrapper > .test')).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
