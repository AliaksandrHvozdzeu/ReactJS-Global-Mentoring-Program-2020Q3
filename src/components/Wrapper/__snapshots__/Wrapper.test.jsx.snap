// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Wrapper rendered then child component should be rendered 1`] = `
<div
  className="wrapper unselectable"
>
  <div
    className="test"
  >
    Test child element
  </div>
</div>
`;
